src/server/pfs/server/obj_block_api_server.go

```diff
@@ -296,12 +296,45 @@ func (s *objBlockAPIServer) InspectObject(ctx context.Context, request *pfsclien
 	return objectInfo, nil
 }
 
+func (s *objBlockAPIServer) ListObjects(request *pfsclient.ListObjectsRequest, listObjectsServer pfsclient.ObjectAPI_ListObjectsServer) (retErr error) {
+	func() { s.Log(request, nil, nil, 0) }()
+	defer func(start time.Time) { s.Log(request, nil, retErr, time.Since(start)) }(time.Now())
+
+	return s.objClient.Walk(s.localServer.objectDir(), func(hash string) error {
+		return listObjectsServer.Send(&pfsclient.Object{hash})
```

Comment:
Sure, I'm just thinking about the latency of sending each object one at a time over the streaming API